### ISSUE TITLE
Broken example in README?

### ISSUE THREAD

**Issue by bnoordhuis** [napi-rs/napi-rs#1158]
The example in the README instructs us to do this:
```rust
#[macro_use]
extern crate napi;

#[napi]
fn foo() {}
```
But for me that fails to compile (this is with rustc 1.60.0, 2021 edition):
```
error: cannot find attribute `napi` in this scope
 --> src/lib.rs:4:3
  |
4 | #[napi]
  |   ^^^^
  |
  = note: consider importing this attribute macro:
          napi_derive::napi
```
And indeed changing it to `use napi_derive::napi;` fixes it (good compiler hint!)

<hr>

A secondary issue is that I get "function is never used" rust-analyzer warnings for functions tagged with `#[napi]` but I can fix that by either making them `pub` or turning on `#[allow(dead_code)]`.

**Comment by Brooooooklyn** [comment]
> many of the examples fail to compile for me with `cargo build --all-features.`

Yes, the `napi-derive` has a `noop` feature to allow developers to disable the code generation. This feature assumes you have a package A, and if you want to share this package A to another non NAPI-RS project like a normal crate, you will need to enable the `noop` feature for it.

But if you pass `--all-features` in this project, the `noop` feature will be activated, and example crates will no longer be compiled.

**Comment by bnoordhuis** [comment]
Thanks for the explanation. I'm trying to ensure I don't break anything when making changes. Is there a good way to test that without rummaging through every Cargo.toml for features?

(I think I count over 30 of them and I'd hate to type them out by hand each time. :-))

**Comment by Brooooooklyn** [comment]
@bnoordhuis you can comment this line:

https://github.com/napi-rs/napi-rs/blob/e7464383d3d9fb3af4a7645499ff840eab944bb0/crates/macro/Cargo.toml#L19

and run the `cargo check --all-features`
